lib/fastlane/plugin/wpmreleasetoolkit/actions/common/publish_github_release_action.rb

```diff
@@ -0,0 +1,63 @@
+require 'fastlane/action'
+require_relative '../../helper/github_helper'
+
+module Fastlane
+  module Actions
+    class PublishGithubReleaseAction < Action
+      def self.run(params)
+        repository = params[:repository]
+        name = params[:name]
+        prerelease = params[:prerelease]
+
+        github_helper = Fastlane::Helper::GithubHelper.new(github_token: params[:github_token])
+
+        url = github_helper.publish_release(
+          repository: repository,
+          name: name,
+          prerelease: prerelease == :unchanged ? nil : prerelease
+        )
+        UI.success("Successfully published GitHub Release #{name}. You can see it at '#{url}'")
+        url
+      end
+
+      def self.description
+        'Publish an existing GitHub Release still in draft mode'
+      end
+
+      def self.authors
+        ['Automattic']
+      end
+
+      def self.return_value
+        'The URL of the published GitHub Release'
+      end
+
+      def self.details
+        'Publish an existing GitHub Release still in draft mode'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :repository,
+                                       description: 'The slug (`<org>/<repo>`) of the GitHub repository we want to create the release on',
+                                       optional: false,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :name,
+                                       description: 'The name (aka title) of the draft release to publish',
+                                       optional: false,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :prerelease,
+                                       description: 'True if this should be published as a pre-release. False if this should be published as final. Don't provide a value keep the same (non-)prerelease status as the one used in the Draft',
```

Comment:
<table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/release-toolkit/blob/6e552e0dc5db3a6e2b04b5e7af166d885bbb6f34/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/publish_github_release_action.rb#L50"></span>Lint/Syntax: unterminated string meets end of file<br />
(Using Ruby 3.2 parser; configure using <code>TargetRubyVersion</code> parameter, under <code>AllCops</code>)</td>
    </tr>
  </tbody>
</table>